packages/oauth/src/stores/file-store.ts

```diff
@@ -0,0 +1,116 @@
+import { Installation, InstallationStore, InstallationQuery } from '../index';
+import { Logger } from '../logger';
+import fs from 'fs';
+import path from 'path';
+import { homedir } from 'os';
+
+export interface FileInstallationOptions {
+  baseDir?: string;
+  historicalDataEnabled?: boolean;
+  clientId?: string;
+}
+
+export class FileInstallationStore implements InstallationStore {
+
+  private baseDir: string;
+  private historicalDataEnabled: boolean;
+
+  constructor({
+    baseDir = `${homedir()}/.bolt-app-installation`,
+    clientId,
+    historicalDataEnabled = true,
+  }: FileInstallationOptions = {}) {
+    this.baseDir = clientId !== undefined ? `${baseDir}/${clientId}` : baseDir;
+    this.historicalDataEnabled = historicalDataEnabled;
+  }
+
+  public async storeInstallation(installation: Installation, logger?: Logger): Promise<void> {
+    const { enterprise, team, user } = installation;
+    const installationData = JSON.stringify(installation);
+    const installationDir = this.getInstallationDir(enterprise?.id, team?.id);
+
+    if (logger !== undefined) { logger.warn('Storing installation in FileInstallationStore'); }
+
+    // Create Installation Directory
+    fs.mkdir(installationDir, { recursive: true }, (err) => {
+      if (err !== null) { return console.error(err); }
+    });
+
+    try {
```

Comment:
I mentioned this in the summary, but this is where it's applicable, so pasting:

> The Python FileStoreInstallation goes a step further by saving the same information into both installer-latest / installer-[currentUTC] and the user-[user_id]-* records. I omitted the former records because it wasn't clear to me how the duplication was value-adding, but am open to introducing it if folks feel it's important or if I've misunderstood its purpose.